amr-wind/wind_energy/ABLBoundaryPlane.cpp

```diff
@@ -0,0 +1,624 @@
+    const int nlevels = m_repo.num_active_levels();
+    for (amrex::OrientationIter oit; oit; ++oit) {
+        auto ori = oit();
+        const std::string plane = m_plane_names[ori];
+
+        if (std::find(m_out_planes.begin(), m_out_planes.end(), plane) ==
+            m_out_planes.end())
+            continue;
+
+        auto plane_grp = ncf.def_group(plane);
+
+        const int normal = ori.coordDir();
+        auto v_normal = plane_grp.def_scalar("normal", NC_INT);
+        v_normal.put(&normal);
+
+        const int face_dir = ori.faceDir();
+        auto v_face = plane_grp.def_scalar("side", NC_INT);
+        v_face.put(&face_dir);
+
+        const amrex::GpuArray<int, 2> perp = get_perpendicular_idx(normal);
+        auto v_perp = plane_grp.def_var("perpendicular", NC_INT, {"pdim"});
+        v_perp.put(perp.data());
+
+        for (int lev = 0; lev < nlevels; ++lev) {
+
+            auto lev_grp = plane_grp.def_group(get_level_name(lev));
+
+            // FIXME remove all c_str
+
+            const amrex::Box& minBox = m_mesh.boxArray(lev).minimalBox();
+            const auto& lo = minBox.loVect();
+            const auto& hi = minBox.hiVect();
+            lev_grp.def_dim("nx", minBox.length(0));
+            lev_grp.def_dim("ny", minBox.length(1));
+            lev_grp.def_dim("nz", minBox.length(2));
+
+            auto v_lengths = lev_grp.def_var("lengths", NC_DOUBLE, {"pdim"});
+            auto v_lo = lev_grp.def_var("lo", NC_DOUBLE, {"pdim"});
+            auto v_hi = lev_grp.def_var("hi", NC_DOUBLE, {"pdim"});
+            auto v_dx = lev_grp.def_var("dx", NC_DOUBLE, {"pdim"});
+
+            const amrex::Vector<std::string> dirs{"nx", "ny", "nz"};
+            for (auto* fld : m_out_fields) {
+                const std::string name = fld->name();
+                if (fld->num_comp() == 1) {
+                    lev_grp.def_var(
+                        name, NC_DOUBLE, {"nt", dirs[perp[0]], dirs[perp[1]]});
+                } else if (fld->num_comp() == AMREX_SPACEDIM) {
+                    lev_grp.def_var(
+                        name, NC_DOUBLE,
+                        {"nt", dirs[perp[0]], dirs[perp[1]], "vdim"});
+                }
+            }
+
+            // Populate coordinates
+            const auto& dx = m_mesh.Geom(lev).CellSizeArray();
+            const amrex::Vector<amrex::Real> pdx{{dx[perp[0]], dx[perp[1]]}};
+            const amrex::Vector<amrex::Real> los{
+                {lo[perp[0]] * dx[perp[0]], lo[perp[1]] * dx[perp[1]]}};
+            const amrex::Vector<amrex::Real> his{
+                {(hi[perp[0]] + 1) * dx[perp[0]],
+                 (hi[perp[1]] + 1) * dx[perp[1]]}};
+            const amrex::Vector<amrex::Real> lengths{
+                {minBox.length(perp[0]) * dx[perp[0]],
+                 minBox.length(perp[1]) * dx[perp[1]]}};
+
+            v_lengths.put(lengths.data());
+            v_lo.put(los.data());
+            v_hi.put(his.data());
+            v_dx.put(pdx.data());
```

Comment:
For NetCDF4 files, this called `nc_enddef` since switched from define mode to data update mode

Reply:
done